File: app/src/main/java/com/android3/siegertpclient/data/game/gamesource/gameLocal/GameLocalDataSource.java
package com.android3.siegertpclient.data.game.gamesource.gameLocal;

import com.android3.siegertpclient.data.game.Game;
import com.android3.siegertpclient.data.invitation.Invitation;
import com.android3.siegertpclient.data.invitation.invitationsource.invitationLocal.InvitationDao;

import org.jetbrains.annotations.NotNull;

import java.util.regex.Pattern;

public class GameLocalDataSource {

        final GameDao gameDao;


    public GameLocalDataSource(GameDao gameDao) {
        this.gameDao = gameDao;
    }

    public void saveGame(@NotNull Game game) {
            gameDao.insertGame(game);
        }

    public boolean check() {

        return false;
    }

    }

File: app/src/main/java/com/android3/siegertpclient/data/game/gamesource/gameLocal/GameLocalDataSource.kt
package com.android3.siegertpclient.data.game.gamesource.gameLocal

import com.android3.siegertpclient.data.Result.Result
import com.android3.siegertpclient.data.game.Game

/**
 * local data source of the room database.
 */
class GameLocalDataSource(val gameDao: GameDao) {
    fun saveGame(game: Game) {
        gameDao.insertGame(game)
    }

    fun getGame(){
        gameDao.allGames
    }
    fun deleteGame(){
        gameDao.deleteAllGames()
    }
    fun updateGame(game: Game){
        gameDao.updateGame(game)
    }
    fun check(): Boolean {
        return true
    }
}